src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -116,6 +118,12 @@ object Instance {
      * @return An instance in the scheduled state.
      */
     def apply(runSpec: RunSpec): Instance = Scheduled(runSpec, Id.forRunSpec(runSpec.id))
+
+    def apply(scheduledInstance: Instance, reservation: Reservation, agentInfo: AgentInfo): Instance = {
```

Comment:
Could you add a doc string? I'm not sure this easier than `instance.copy` 😉

Reply:
well to move to a certain instance states, you have to know which fields to copy. Having this knowledge spread over multiple places in the codebase is brittle - if you need to start modifying one more field, you can easily forget to change it at all places

Reply:
Good point.